### PR TITLE
Use NumberInput for Image Run Dialog

### LINKED ISSUE #858: Use NumberInput for Image Run Dialog
Use the [number input](https://www.patternfly.org/v4/components/number-input) in the Image Create modal to replace the issues with the current input type=number.

### PR DESCRIPTION
**Pull request by marusak**
Fixes #858

### CODE REVIEW

**Comment by martinpitt** [comment]
Needs more pixel updates

File: src/ImageRunModal.jsx [hunk 1]
```diff
@@ -80,24 +80,30 @@ const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
...
-                            step={1}
-                            min={1}
-                            max={65535}
-                            value={item.containerPort || ''}
-                            onChange={value => onChange(idx, 'containerPort', value)} />
+                <NumberInput
+                        id={id + "-container-port"}
+                        min={1}
+                        max={65535}
+                        value={item.containerPort || ''}
```
Comment (jelly) [inline src/ImageRunModal.jsx:101]: Numberinput's value should be a number

File: src/ImageRunModal.jsx [hunk 2]
```diff
@@ -80,24 +80,30 @@ const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
...
-                            step={1}
-                            min={1}
-                            max={65535}
-                            value={item.hostPort || ''}
-                            onChange={value => onChange(idx, 'hostPort', value)} />
+                <NumberInput
+                        id={id + "-host-port"}
+                        min={1}
+                        max={65535}
+                        value={item.hostPort || ''}
```
Comment (jelly) [inline src/ImageRunModal.jsx:87]: Same as below

File: src/ImageRunModal.jsx (hunk 1 above)
Comment (marusak) [reply src/ImageRunModal.jsx:101]: Then we would need to default to 0 and I don't like that behavior. See in https://www.patternfly.org/v4/components/number-input Let's assume I want to enter there value `5`.

1. Select the current value
2. Delete it
3. It now has 0
4. Enter 5
5. It now has `05`. It is not possible to enter your own value without leading 0

Comment (marusak) [reply src/ImageRunModal.jsx:101]: I know that it is defined as `number` but empty string just works :man_shrugging:

Comment (jelly) [reply src/ImageRunModal.jsx:101]: As long as it doesn't give a console error it's fine I suppoes. But yeah there is no good default host/port for podman containers.

Comment (marusak) [reply src/ImageRunModal.jsx:101]: it does not give any error in console

**Comment by marusak** [comment]
@garrett I have dropped those ports. Now it is used at 3 places - cpu shares, memory limit and number of retries in restart policy.

**Comment by garrett** [comment]
Better, but I was still able to scroll to negative and create a container with this:

Screenshot 2022-05-31 at 11-21-43 Podman containers - garrett@Bolt

Screenshot 2022-05-31 at 11-26-24 Podman containers - garrett@Bolt

**Review by garrett — APPROVED** [review]
Better, but I was still able to scroll to negative and create a container with this:

Screenshot 2022-05-31 at 11-21-43 Podman containers - garrett@Bolt

Screenshot 2022-05-31 at 11-26-24 Podman containers - garrett@Bolt

I'm happy with the NumberInput now, but please fix this issue in a follow-up.

**Comment by marusak** [comment]
Hmm, there is defined `min=0` but I will take a look

**Comment by marusak** [comment]
@garrett could you please take a look?

**Comment by garrett** [comment]
I see very odd behavior with the + button when a number is entered manually:

https://user-images.githubusercontent.com/10246/172569862-a7c2383f-850e-4ee8-8bd3-c2efa2b9c551.mp4

**Review by garrett — CHANGES_REQUESTED** [review]
I see very odd behavior with the + button when a number is entered manually:

https://user-images.githubusercontent.com/10246/172569862-a7c2383f-850e-4ee8-8bd3-c2efa2b9c551.mp4

**Comment by marusak** [comment]
> I see very odd behavior with the + button when a number is entered manually:

right, thanks, fixed!

File: src/ImageRunModal.jsx [hunk 3]
```diff
@@ -935,15 +935,17 @@ export class ImageRunModal extends React.Component {
...
-                                   step={1}
                                    min={0}
-                                   isReadOnly={!this.state.memoryConfigure}
+                                   isDisabled={!this.state.memoryConfigure}
                                    onClick={() => !this.state.memoryConfigure && this.onValueChanged('memoryConfigure', true)}
-                                   onChange={value => this.onValueChanged('memory', value)} />
+                                   onPlus={() => this.onPlusOne('memory')}
+                                   onMinus={() => this.onMinusOne('memory')}
+                                   minusBtnAriaLabel={_("Decrease memory by one")}
+                                   plusBtnAriaLabel={_("Increase memory by one")}
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:947]: "by one" sounds strange -- if we can't easily add the unit here, maybe just drop the "by one"?

File: src/ImageRunModal.jsx [hunk 4]
```diff
@@ -974,15 +976,18 @@ export class ImageRunModal extends React.Component {
...
-                                        step={1}
                                         min={2}
                                         max={262144}
-                                        isReadOnly={!this.state.cpuSharesConfigure}
-                                        onChange={value => this.onValueChanged('cpuShares', parseInt(value))} />
+                                        isDisabled={!this.state.cpuSharesConfigure}
+                                        onPlus={() => this.onPlusOne('cpuShares')}
+                                        onMinus={() => this.onMinusOne('cpuShares')}
+                                        minusBtnAriaLabel={_("Decrease cpu shares by one")}
+                                        plusBtnAriaLabel={_("Increase cpu shares by one")}
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:989]: "CPU shares"

File: src/ImageRunModal.jsx [hunk 5]
```diff
@@ -543,11 +543,11 @@ export class ImageRunModal extends React.Component {
     }
 
     onPlusOne(key) {
-        this.setState(state => ({ [key]: state[key] + 1 }));
+        this.setState(state => ({ [key]: Number(state[key]) + 1 }));
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:546]: This PR uses parseInt() and Number() -- what's the main difference here?

File: src/ImageRunModal.jsx [hunk 6]
```diff
@@ -1083,11 +1090,11 @@ export class ImageRunModal extends React.Component {
                                         min={0}
                                         max={262144}
                                         widthChars={6}
-                                        minutBtnAriaLabel={_("Decrease interval by one")}
+                                        minusBtnAriaLabel={_("Decrease interval by one")}
                                         plusBtnAriaLabel={_("Increase interval by one")}
                                         onMinus={() => this.onMinusOne('healthcheck_interval')}
                                         onPlus={() => this.onPlusOne('healthcheck_interval')}
-                                        onChange={ev => this.onValueChanged('healthcheck_interval', parseInt(ev.target.value))} />
+                                        onChange={ev => this.onValueChanged('healthcheck_interval', Number(ev.target.value) < 0 ? 0 : ev.target.value)} />
```
Comment (martinpitt) [inline src/ImageRunModal.jsx:1097]: i.e. what's the practical effect of changing parseInt to Number here?

**Review by garrett — DISMISSED** [review]
Thanks! :+1:  LGTM

File: src/ImageRunModal.jsx (hunk 3 above)
Comment (marusak) [reply src/ImageRunModal.jsx]: removed

File: src/ImageRunModal.jsx (hunk 5 above)
Comment (marusak) [reply src/ImageRunModal.jsx]: Switched to using parseInt everywhere

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**